web3/utils/transactions.py

```diff
@@ -168,3 +180,37 @@ def get_buffered_gas_estimate(web3, transaction, gas_buffer=100000):
         )
 
     return min(gas_limit, gas_estimate + gas_buffer)
+
+
+def extract_valid_transaction_params(transaction_params):
+    return { key: transaction_params[key] for key in VALID_TRANSACTION_PARAMS if key in transaction_params }
+
+
+def assert_valid_transaction_params(transaction_params):
+    for param in transaction_params:
+        if param not in VALID_TRANSACTION_PARAMS:
+            raise ValueError('{} is not a valid transaction parameter'.format(param))
+
+
+def prepare_replacement_transaction(web3, current_transaction, new_transaction):
+    if current_transaction['blockHash'] is not None:
+        raise ValueError('Supplied transaction with hash {} has already been mined'
+                         .format(current_transaction['hash']))
+    if 'nonce' in new_transaction and new_transaction['nonce'] != current_transaction['nonce']:
+        raise ValueError('Supplied nonce in new_transaction must match the pending transaction')
+
+    if 'nonce' not in new_transaction:
+        new_transaction = assoc(new_transaction, 'nonce', current_transaction['nonce'])
+
+    if 'gasPrice' in new_transaction:
+        if new_transaction['gasPrice'] < current_transaction['gasPrice']:
```

Comment:
The replacement must be strictly greater than current, so this check should be `<=`